extmod/nimble/nimble/nimble_npl_os.c

```diff
@@ -56,65 +57,73 @@ void *ble_npl_get_current_task_id(void) {
 /******************************************************************************/
 // malloc
 
-// Maintain a linked list of heap memory that we've passed to Nimble,
-// discoverable via the bluetooth_nimble_memory root pointer.
+// Forwards directly to GC heap allocations, however we mark them as permanent
+// so that they survive soft resets.
 
-typedef struct _mp_bluetooth_nimble_malloc_t {
-    struct _mp_bluetooth_nimble_malloc_t *prev;
-    struct _mp_bluetooth_nimble_malloc_t *next;
+#if !MICROPY_GC_PERMANENT_ALLOCS
+#error nimble_npl_os requires MICROPY_GC_PERMANENT_ALLOCS
+#endif
+
+#if MICROPY_MALLOC_USES_ALLOCATED_SIZE
+
+typedef struct _sized_malloc_t {
     size_t size;
     uint8_t data[];
-} mp_bluetooth_nimble_malloc_t;
+} sized_malloc_t;
 
-// TODO: This is duplicated from mbedtls. Perhaps make this a generic feature?
 STATIC void *m_malloc_bluetooth(size_t size) {
-    size += sizeof(mp_bluetooth_nimble_malloc_t);
-    mp_bluetooth_nimble_malloc_t *alloc = m_malloc0(size);
+    size += sizeof(sized_malloc_t);
+
+    sized_malloc_t *alloc = m_malloc(size);
```

Comment:
this used to be a `m_malloc0`, does that matter?

Reply:
This to clear the linked list members.

Reply:
ok